engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/UpdateByWindowTime.java

```diff
@@ -0,0 +1,400 @@
+                final LongColumnIterator influencerTsHeadIt = new LongColumnIterator(context.timestampColumnSource, context.influencerRows);
+                final LongColumnIterator influencerTsTailIt = new LongColumnIterator(context.timestampColumnSource, context.influencerRows);
+                final RowSequence.Iterator influencerKeyIt = ctx.influencerRows.getRowSequenceIterator();
+                final ChunkSource.GetContext localTimestampContext =
+                        ctx.timestampColumnSource.makeGetContext(ctx.workingChunkSize);
+                final WritableIntChunk<? extends Values> pushChunk =
+                        WritableIntChunk.makeWritableChunk(ctx.workingChunkSize);
+                final WritableIntChunk<? extends Values> popChunk =
+                        WritableIntChunk.makeWritableChunk(ctx.workingChunkSize)) {
+
+            long currentHeadTs = advanceToNonNull(influencerTsHeadIt);
+            long currentTailTs = advanceToNonNull(influencerTsTailIt);
+
+            while (it.hasMore()) {
+                final RowSequence chunkRs = it.getNextRowSequenceWithLength(ctx.workingChunkSize);
+                final int chunkRsSize = chunkRs.intSize();
+
+                // NOTE: we did not put null values into our SSA and our influencer rowset is built using the
+                // SSA. there should be no null timestamps considered in the rolling windows
+                final LongChunk<? extends Values> timestampChunk =
+                        ctx.timestampColumnSource.getChunk(localTimestampContext, chunkRs).asLongChunk();
+
+                // chunk processing
+                long totalCount = 0;
+
+                for (int ii = 0; ii < chunkRsSize; ii++) {
+                    // read the current timestamp
+                    final long currentTimestamp = timestampChunk.get(ii);
+                    if (currentTimestamp == NULL_LONG) {
+                        // this signifies that does not belong to a time window
+                        popChunk.set(ii, NULL_INT);
+                        pushChunk.set(ii, NULL_INT);
+                        continue;
+                    }
+
+                    // compute the head and tail timestamps (inclusive)
+                    final long head = currentTimestamp - prevUnits;
+                    final long tail = currentTimestamp + fwdUnits;
+
+                    // pop out all values from the current window that are not in the new window
+                    long popCount = 0;
+                    while (currentHeadTs < head) {
+                        currentHeadTs = advanceToNonNull(influencerTsHeadIt);
+                        popCount++;
+                    }
+
+                    // push in all values that are in the new window (inclusive of tail)
+                    long pushCount = 0;
+                    while (currentTailTs <= tail) {
+                        currentTailTs = advanceToNonNull(influencerTsTailIt);
```

Comment:
We're iterating the TS data twice. And not caching it. We should think about this.

Reply:
OBE, restored use of `LongRingBuffer` for this case